pytoda/smiles/transforms.py

```diff
@@ -3,24 +3,164 @@
 import re
 
 import numpy as np
+import pytoda
 import torch
 from rdkit import Chem, DataStructs
 from rdkit.Chem import AllChem
-from selfies import encoder
-
-import pytoda
+from selfies import encoder as selfies_encoder
 
-from ..transforms import Transform
-from ..types import Indexes
-from .smiles_language import SMILESLanguage
+from ..transforms import (Compose, LeftPadding, Randomize, StartStop, ToTensor,
+                          Transform)
+from ..types import Callable, Indexes, Tensor, Tuple, Union
 
 logger = logging.getLogger('pytoda_SMILES_transforms')
 
 
+def compose_smiles_transforms(
+    canonical: bool = False,
+    augment: bool = False,
+    kekulize: bool = False,
+    all_bonds_explicit: bool = False,
+    all_hs_explicit: bool = False,
+    remove_bonddir: bool = False,
+    remove_chirality: bool = False,
+    selfies: bool = False,
+    sanitize: bool = True,
+) -> Compose:
+    """Setup a composition of SMILES to SMILES (or SELFIES) transformations.
+
+    Args:
+        canonical (bool, optional): performs canonicalization of SMILES
+        (one original string for one molecule). If True, then other
+        transformations (augment etc, see below) do not apply. Defaults to
+            False.
+        augment (bool, optional): perform SMILES augmentation. Defaults to
+            False.
+        kekulize (bool, optional): kekulizes SMILES (implicit aromaticity
+            only). Defaults to False.
+        all_bonds_explicit (bool, optional): makes all bonds explicit.
+            Defaults to False, only applies if kekulize is True.
+        all_hs_explicit (bool, optional): makes all hydrogens explicit.
+            Defaults to False, only applies if kekulize is True.
+        remove_bonddir (bool, optional): remove directional info of bonds.
+            Defaults to False.
+        remove_chirality (bool, optional): remove chirality information.
+            Defaults to False.
+        selfies (bool, optional): whether selfies is used instead of
+            smiles. Defaults to False.
+        sanitize (bool, optional): sanitize SMILES. Defaults to True.
+
+    Returns:
+        Compose: A Callable that applies composition of SMILES transforms.
+    """
+    # TODO improve sanitize docstring
+    # ?? Kekulize, check valencies, set aromaticity, conjugation and hybridization
```

Comment:
as far as I understand, we don't necessarily need more in depth checks on the molecules. Most of the transforms don't seem to have major impact on the QSAR models

Reply:
The comment was respective to the sanitize docstring. It was from the time I asked you if sanitize would always also kekulize because I saw a function in rdkit do so. Personally I'm still not sure what exactly sanitize does but it's fair to refer to rdkit.

Reply:
It's extremely obscure what sanitize does, rdkit documentation is not even vague but just gives no info at all. Especially if the backend is in C, developers should really make an effort to document their code...

The best we can do is delegating to rdkit I think